app_test.go

```diff
@@ -473,3 +493,66 @@ func TestNopLogger(t *testing.T) {
 	app := fxtest.New(t, NopLogger)
 	app.RequireStart().RequireStop()
 }
+
+type testWrappedErr struct{}
+
+func (we testWrappedErr) Graph() string {
+	return "graph"
+}
+
+func (we testWrappedErr) Error() string {
+	return "great sadness"
+}
+
+func TestVisualizeError(t *testing.T) {
+	t.Run("NotWrappedError", func(t *testing.T) {
+		err := fmt.Errorf("great sadness")
+		assert.Equal(t, "", VisualizeError(err))
+	})
+
+	t.Run("WrappedError", func(t *testing.T) {
+		err := testWrappedErr{}
+		assert.Equal(t, "graph", VisualizeError(err))
+	})
+}
+
+var errStr, graphStr string
```

Comment:
I'm never a fan of globals, even in tests. Leaving globals around may cause future tests to rely on previous behviour which is tricky to debug.

What if we made these fields on `graphErrHandler` and do the assertions directly on the `errHandler` you define?